readthedocs/projects/models.py

```diff
@@ -119,6 +117,18 @@ def save(self, *args, **kwargs):  # pylint: disable=arguments-differ
     def get_absolute_url(self):
         return resolve(self.child)
 
+    @cached_property
+    def subproject_prefix(self):
+        """
+        Returns the path prefix of the subproject.
+
+        This normally is ``/projects/<subproject-alias>/``,
+        but if the project has a custom subproject prefix,
+        that will be used.
+        """
+        prefix = self.parent.custom_subproject_prefix or "/projects/"
+        return unsafe_join_url_path(prefix, self.alias, "/")
```

Comment:
Does this require the prefix to start with a `/`? We should make this logic as hard to mess up as possible, so would be good to reduce requirements on the prefix.

Reply:
All prefixes are normalized to start and end with one slash.